### PR TITLE
Add SharedArrayBuffer and Atomics compat data

### PR DESCRIPTION
**Pull request by Elchi3**

### CODE REVIEW

File: javascript/builtins/Atomics.json [hunk 1]
```diff
@@ -0,0 +1,708 @@
...
+              "version_added": false
+            },
+            "chrome": {
+              "version_added": "60"
+            },
+            "chrome_android": {
+              "version_added": false
+            },
+            "edge": {
+              "version_added": false
```
Comment (wbamberg) [inline javascript/builtins/Atomics.json]: 16 according to the page

File: javascript/builtins/Atomics.json [hunk 2]
```diff
@@ -0,0 +1,708 @@
...
+        },
+        "add": {
+          "__compat": {
+            "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Atomics/add",
+            "support": {
+              "webview_android": {
+                "version_added": false
+              },
+              "chrome": {
+                "version_added": "60"
```
Comment (wbamberg) [inline javascript/builtins/Atomics.json]: false/needs runtime flags according to the page

File: javascript/builtins/Atomics.json [hunk 3]
```diff
@@ -0,0 +1,708 @@
...
+                "version_added": false
+              },
+              "edge": {
+                "version_added": false
+              },
+              "edge_mobile": {
+                "version_added": false
+              },
+              "firefox": {
+                "version_added": "55"
```
Comment (wbamberg) [inline javascript/builtins/Atomics.json]: behind a preference 46-54 according to the page

File: javascript/builtins/Atomics.json [hunk 4]
```diff
@@ -0,0 +1,708 @@
...
+            "status": {
+              "experimental": false,
+              "standard_track": true,
+              "deprecated": false
+            }
+          }
+        },
+        "and": {
+          "__compat": {
+            "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Atomics/and",
```
Comment (wbamberg) [inline javascript/builtins/Atomics.json]: same comments as for add

File: javascript/builtins/Atomics.json [hunk 5]
```diff
@@ -0,0 +1,708 @@
...
+            "status": {
+              "experimental": false,
+              "standard_track": true,
+              "deprecated": false
+            }
+          }
+        },
+        "compareExchange": {
+          "__compat": {
+            "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Atomics/compareExchange",
```
Comment (wbamberg) [inline javascript/builtins/Atomics.json]: same comments as for add. I'm going to stop noting these ones now.

File: javascript/builtins/Atomics.json [hunk 6]
```diff
@@ -0,0 +1,708 @@
...
+                "version_added": false
+              },
+              "opera": {
+                "version_added": false
+              },
+              "opera_android": {
+                "version_added": false
+              },
+              "safari": {
+                "version_added": "10.1"
```
Comment (wbamberg) [inline javascript/builtins/Atomics.json]: false according to the page

File: javascript/builtins/Atomics.json [hunk 7]
```diff
@@ -0,0 +1,708 @@
...
```
Comment (wbamberg) [inline javascript/builtins/Atomics.json]: false according to the page

File: javascript/builtins/Atomics.json [hunk 8]
```diff
@@ -0,0 +1,708 @@
...
```
Comment (wbamberg) [inline javascript/builtins/Atomics.json]: false according to the page. I'm going to stop noting these ones now.

File: javascript/builtins/Atomics.json [hunk 9]
```diff
@@ -0,0 +1,708 @@
...
+              "chrome_android": {
+                "version_added": false
+              },
+              "edge": {
+                "version_added": false
+              },
+              "edge_mobile": {
+                "version_added": false
+              },
+              "firefox": {
```
Comment (wbamberg) [inline javascript/builtins/Atomics.json]: naming note for versions 46-7 in the page

File: javascript/builtins/Atomics.json [hunk 10]
```diff
@@ -0,0 +1,708 @@
...
+        },
+        "wake": {
+          "__compat": {
+            "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Atomics/wake",
+            "support": {
+              "webview_android": {
+                "version_added": false
+              },
+              "chrome": {
+                "version_added": "60"
```
Comment (wbamberg) [inline javascript/builtins/Atomics.json]: Extra note for Chrome in the page.

File: javascript/builtins/Atomics.json [hunk 11]
```diff
@@ -0,0 +1,708 @@
...
+                "version_added": false
+              },
+              "edge": {
+                "version_added": false
+              },
+              "edge_mobile": {
+                "version_added": false
+              },
+              "firefox": {
+                "version_added": "55"
```
Comment (wbamberg) [inline javascript/builtins/Atomics.json]: naming note for versions 46-7 in the page

File: javascript/builtins/SharedArrayBuffer.json [hunk 12]
```diff
@@ -0,0 +1,222 @@
...
+    "builtins": {
+      "SharedArrayBuffer": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer",
+          "support": {
+            "webview_android": {
+              "version_added": false
+            },
+            "chrome": {
+              "version_added": "60"
```
Comment (wbamberg) [inline javascript/builtins/SharedArrayBuffer.json]: note in the page

File: javascript/builtins/SharedArrayBuffer.json [hunk 13]
```diff
@@ -0,0 +1,222 @@
...
+            "chrome_android": {
+              "version_added": false
+            },
+            "edge": {
+              "version_added": false
+            },
+            "edge_mobile": {
+              "version_added": false
+            },
+            "firefox": {
```
Comment (wbamberg) [inline javascript/builtins/SharedArrayBuffer.json]: note in the page

File: javascript/builtins/SharedArrayBuffer.json [hunk 14]
```diff
@@ -0,0 +1,222 @@
+{
+  "javascript": {
+    "builtins": {
+      "SharedArrayBuffer": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer",
```
Comment (wbamberg) [inline javascript/builtins/SharedArrayBuffer.json]: the table has 2 subfeatures, which are not included here. Weirdly, the subfeatures have lower version-added than the basic support.

File: javascript/builtins/SharedArrayBuffer.json [hunk 15]
```diff
@@ -0,0 +1,222 @@
...
+        },
+        "prototype": {
+          "__compat": {
+            "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer/prototype",
+            "support": {
+              "webview_android": {
+                "version_added": false
+              },
+              "chrome": {
+                "version_added": "60"
```
Comment (wbamberg) [inline javascript/builtins/SharedArrayBuffer.json]: no support according to the page

File: javascript/builtins/SharedArrayBuffer.json [hunk 16]
```diff
@@ -0,0 +1,222 @@
...
+              "chrome_android": {
+                "version_added": false
+              },
+              "edge": {
+                "version_added": false
+              },
+              "edge_mobile": {
+                "version_added": false
+              },
+              "firefox": {
```
Comment (wbamberg) [inline javascript/builtins/SharedArrayBuffer.json]: note in the page

File: javascript/builtins/SharedArrayBuffer.json [hunk 17]
```diff
@@ -0,0 +1,222 @@
...
+        },
+        "byteLength": {
+          "__compat": {
+            "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer/byteLength",
+            "support": {
+              "webview_android": {
+                "version_added": false
+              },
+              "chrome": {
+                "version_added": "60"
```
Comment (wbamberg) [inline javascript/builtins/SharedArrayBuffer.json]: no support in the page

File: javascript/builtins/SharedArrayBuffer.json [hunk 18]
```diff
@@ -0,0 +1,222 @@
...
+              "chrome_android": {
+                "version_added": false
+              },
+              "edge": {
+                "version_added": false
+              },
+              "edge_mobile": {
+                "version_added": false
+              },
+              "firefox": {
```
Comment (wbamberg) [inline javascript/builtins/SharedArrayBuffer.json]: note in the page

File: javascript/builtins/SharedArrayBuffer.json [hunk 19]
```diff
@@ -0,0 +1,222 @@
...
+        },
+        "slice": {
+          "__compat": {
+            "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer/slice",
+            "support": {
+              "webview_android": {
+                "version_added": false
+              },
+              "chrome": {
+                "version_added": "60"
```
Comment (wbamberg) [inline javascript/builtins/SharedArrayBuffer.json]: no support in the page. I'll stop noting this now.

File: javascript/builtins/SharedArrayBuffer.json [hunk 20]
```diff
@@ -0,0 +1,222 @@
...
+              "chrome_android": {
+                "version_added": false
+              },
+              "edge": {
+                "version_added": false
+              },
+              "edge_mobile": {
+                "version_added": false
+              },
+              "firefox": {
```
Comment (wbamberg) [inline javascript/builtins/SharedArrayBuffer.json]: note in the page. I'll stop noting this now.

File: javascript/builtins/Atomics.json (hunk 1 above)
Comment (Elchi3) [reply javascript/builtins/Atomics.json]: The page got updated in the meantime! :speak_no_evil: I will include this.

File: javascript/builtins/Atomics.json (hunk 2 above)
Comment (Elchi3) [reply javascript/builtins/Atomics.json]: It's shipped in Chrome 60 according to https://www.chromestatus.com/features/4570991992766464. Updating webview_android, and chrome_android, too.

File: javascript/builtins/Atomics.json (hunk 3 above)
Comment (Elchi3) [reply javascript/builtins/Atomics.json]: Yeah, I forgot about this. I think, now that there is a version without preference, most people are only interested in that version (55 here). I hope the {{compact_compat}} (or however we will call the summary compat info at the top of pages), will remove preffed version ranges.

File: javascript/builtins/Atomics.json (hunk 7 above)
Comment (Elchi3) [reply javascript/builtins/Atomics.json]: The subpages haven't been updated very likely. The whole API should be in 10.1 like stated on the main SharedArrayBuffer page https://webkit.org/blog/7477/new-web-features-in-safari-10-1/

File: javascript/builtins/SharedArrayBuffer.json (hunk 14 above)
Comment (Elchi3) [reply javascript/builtins/SharedArrayBuffer.json]: `slice` is there and I'm adding SAB in DataView. The lower versions come through the preference situation, having the preference data will make it more clear.

File: javascript/builtins/SharedArrayBuffer.json (hunk 12 above)
Comment (Elchi3) [reply javascript/builtins/SharedArrayBuffer.json]: Page is outdated.

**Comment by Elchi3** [comment]
> It's not clear to me when discrepancies are intentional and when they are not.
> It would be helpful if these PRs had a description explaining places where the data intentionally diverges from the pages. As it is I feel I'm noting a lot of stuff you already know :-/

I'm sorry about this. I tried to look at the whole API here which should have the same compat data. Unfortunately, on MDN, only the main pages were updated leaving the subpages with outdated compat info. Then I tried to also put in correct data that I know from other sources. Then someone edited the MDN pages after I proposed this PR and you reviewed it. So there were multiple things that made this hard. Some things I also screwed up in my initial migration. Thanks for catching some of these.

In the future, I'll try to be close to the MDN tables and if I diverge I will make a note when I open a PR. Thanks for your time to review this.

**Pull request merged**